it/src/main/java/com/google/cloud/teleport/it/spanner/SpannerResourceManagerUtils.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it.spanner;
+
+import static com.google.common.hash.Hashing.goodFastHash;
+
+import com.google.common.base.CharMatcher;
+import com.google.common.hash.HashFunction;
+import com.google.re2j.Matcher;
+import com.google.re2j.Pattern;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Random;
+
+/**
+ * Utilities for {@link com.google.cloud.teleport.it.spanner.SpannerResourceManager}
+ * implementations.
+ */
+public final class SpannerResourceManagerUtils {
+  private static final DateTimeFormatter TIME_FORMAT =
+      DateTimeFormatter.ofPattern("yyyyMMdd-HHmmss-SSSSSS");
+
+  private SpannerResourceManagerUtils() {}
+
+  /**
+   * Generates a new id string from an existing one.
+   *
+   * @param id The id string to generate a new id from.
+   * @param targetLength The length of the new id to generate. Must be greater than 8.
```

Comment:
Sorry, missed this: It doesn't look like the restriction on being greater than eight is ever enforced.